waffle-doppelganger/test/integrations/proxied.ts

```diff
@@ -0,0 +1,30 @@
+import {use, expect} from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import {deployContract, MockProvider} from 'ethereum-waffle';
+
+import {doppelganger} from '../../src';
+import Counter from '../helpers/interfaces/Counter.json';
+import Cap from '../helpers/interfaces/Cap.json';
+
+use(chaiAsPromised);
+
+describe('Doppelganger - Integration (called by other contract)', () => {
+  const [wallet] = new MockProvider().getWallets();
+
+  it('mocking mechanism works', async () => {
+    const mockCounter = await doppelganger(wallet, Counter.interface);
+    const capContract = await deployContract(
+      wallet,
+      Cap,
+      [mockCounter.address]
+    );
+
+    await mockCounter.mock.read.returns(5);
+    const ret1 = await expect(capContract.read()).to.be.eventually.fulfilled;
```

Comment:
This syntax`.to.be.eventually.fulfilled` seems outdated, why not just:
```ts
const result = await capContract.read();
expect(result.toNumber()).to.eq(10);
```
#depsyntax